tests/test_enterprise/api/test_views.py

```diff
@@ -767,6 +774,308 @@ def test_post_pending_enterprise_customer_user_logged_out(self):
+    """
+    Test PendingEnterpriseCustomerUserViewSet and LinkLearnersSerializer
+    """
+
+    def create_ent_user(self, user_exists, ecu_exists, pending_ecu_exists, user_email, enterprise_customer):
+        """
+        Creates enterprise users or pending users
+        """
+        user = None
+        if user_exists:
+            user = factories.UserFactory(email=user_email)
+            if ecu_exists:
+                factories.EnterpriseCustomerUserFactory(user_id=user.id, enterprise_customer=enterprise_customer)
+
+        if pending_ecu_exists:
+            factories.PendingEnterpriseCustomerUserFactory(
+                user_email=user_email, enterprise_customer=enterprise_customer
+            )
+        return user
+
+    def setup_admin_user(self):
+        """
+        Creates an admin user and logs them in
+        """
+        client_username = 'client_username'
+        self.client.logout()
+        self.create_user(username=client_username, password=TEST_PASSWORD)
+        self.client.login(username=client_username, password=TEST_PASSWORD)
+
+    @ddt.data(
+        {'user_exists': True, 'ecu_exists': False, 'pending_ecu_exists': True, 'status_code': 201},
+        {'user_exists': False, 'ecu_exists': False, 'pending_ecu_exists': False, 'status_code': 201},
+        {'user_exists': True, 'ecu_exists': False, 'pending_ecu_exists': False, 'status_code': 201},
+    )
+    @ddt.unpack
+    def test_post_pending_enterprise_customer_user_creation(
+            self,
+            user_exists,
+            ecu_exists,
+            pending_ecu_exists,
+            status_code):
+        """
+        Make sure service users can post new PendingEnterpriseCustomerUsers.
+        """
+
+        # create user making the request
+        self.setup_admin_user()
+
+        # Create fake enterprise
+        ent_uuid = fake.uuid4()
+        enterprise_customer = factories.EnterpriseCustomerFactory(uuid=ent_uuid)
+        # Fake enterprise admin permissions
+        self.set_jwt_cookie(ENTERPRISE_ADMIN_ROLE, ent_uuid)
+
+        new_user_email = 'newuser@example.com'
+        # data to be passed to the request
+        data = {
+            'enterprise_customer': ent_uuid,
+            'user_email': new_user_email,
+        }
+
+        # create preexisting user(s) as necessary
+        user = self.create_ent_user(
+            user_exists=user_exists,
+            ecu_exists=ecu_exists,
+            pending_ecu_exists=pending_ecu_exists,
+            user_email=new_user_email,
+            enterprise_customer=enterprise_customer,
+        )
+
+        response = self.client.post(
+            settings.TEST_SERVER + reverse('link-pending-enteprise-learner', kwargs={'enterprise_uuid': ent_uuid}),
+            data=data
+        )
+        assert response.status_code == status_code
+        response = self.load_json(response.content)
+        self.assertDictEqual(data, response)
+        if not user_exists:
```

Comment:
Did we want to start a trend where assertions within "if" statements are discouraged?  Is this easier to test as a few separate tests?

Reply:
Heh, this is *after* I took some ifs out of here. It's slightly annoying to do as separate tests. OH. WAIT. I HAVE IDEAS.

Reply:
Nevermind, my ideas did not pan out. This one is trickier to pull apart; when I went through them the first time, I decided it wasn't worth it for this one.